### ISSUE TITLE
registrationId has to be hardwired to '{registrationId}'

### ISSUE THREAD

**Issue by pshotton** [grails-spring-security-saml/grails-spring-security-saml#8]
SpringSecuritySamlGrailsPlugin.groovy has the following check in the doWithSpring closure:
`                if (metadataUrl) {
                    if (!metadataUrl.contains("{registrationId}")) {
                        throw new IllegalArgumentException("grails.plugin.springsecurity.saml.metadata.url must contain {registrationId}")
                    }
                    requestMatcher = new AntPathRequestMatcher(metadataUrl)
                }
  `

thus requiring the metadataUrl to be hardwired to '{registrationId}'.

Should the test be comparing "${registrationId}"?

There's a couple of other locations where the $ is missing too.

Using Grails 6.2, plugin version 5.0.0-RC3

**Comment by valentingoebel** [comment]
No, this is intentional. It's a Spring Security (or perhaps even Spring proper) aspect of the AntPathRequestMatcher. If you use that `{registrationId}` template, it will parse the registrationId from the URL.

The URL `/login/saml2/sso/abc` with the template `/login/saml2/sso/{registrationId}` gives you `registrationId=abc`

If you want a metadata url like ``/custom-metadata-url`` without the registrationId, then you will have to define your own Saml2MetadataFilter with a custom RequestMatcher that has the hard wired registrationId.

Meanwhile for the SAML SSO and SLO endpoints, there is the `defaultIdp` setting that allows you to set a custom ``assertionConsumerService`` and ``singleLogoutService`` without any restrictions.

**Comment by pshotton** [comment]
Got it. Thanks for your help.
